java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java

```diff
@@ -23,67 +23,24 @@
   public String teamName;            // either "<name>" or "!!<path>"
   public String dataSignerKey;       // should be "subzerodatasigner", unless a different name was given to the key.
 
-  /**
-   * SubzeroGui does not use ServiceContainer, so we implement our own config loading.
-   */
-  public static SubzeroConfig load(boolean nCipher) throws IOException {
-    // load subzero.yaml
-    EnvironmentsMap environmentsMap = loadEnvMap();
-
-    // Get HSM's security world. Default to dev.
-    EnvironmentsMap.Environments env = null;
+  public static SubzeroConfig load(boolean nCipher, String configFile)
+      throws IOException, NFException {
     if (nCipher) {
       String securityWorld = new NCipher().getSecurityWorld();
-      System.out.println(format("Looking up env with %s", securityWorld));
-      env = environmentsMap.environments.get(securityWorld);
+      System.out.println(format("Security world: %s", securityWorld));
     } else {
-      System.out.println("skipping nCipher check");
-    }
-    if (env == null) {
-      env = EnvironmentsMap.Environments.development;
+      System.out.println("skipping nCipher");
     }
-    System.out.printf("Env: %s\n", env.name());
 
     // Load the right config file
-    ClassLoader classLoader = SubzeroGui.class.getClassLoader();
-    URL resource = classLoader.getResource(format("subzero-%s.yaml", env.name()));
-    return new Yaml().loadAs(resource.openStream(), SubzeroConfig.class);
-  }
-
-  protected static EnvironmentsMap loadEnvMap() throws IOException {
-    ClassLoader classLoader = SubzeroGui.class.getClassLoader();
-    URL resource = classLoader.getResource("subzero.yaml");
-    return new Yaml().loadAs(resource.openStream(), EnvironmentsMap.class);
-  }
-
-  private String getStringOrLoadFile(String parameter) throws IOException {
-    if (parameter.startsWith(FILE_PREFIX)) {
-      List<String> fileContent =
-          Files.readAllLines(Paths.get(parameter.substring(FILE_PREFIX.length())));
-      if (fileContent.size() > 0) {
-        return fileContent.get(0);
-      }
-    }
-    return parameter;
-  }
-
-  public String getSoftcardPassword() throws IOException {
-    return getStringOrLoadFile(softcardPassword);
-  }
-
-  public String getTeamName() throws IOException {
-    return getStringOrLoadFile(teamName);
-  }
-
-  public String getFramebufferSize() throws IOException {
-    return getStringOrLoadFile(framebufferSize);
-  }
-
-  public static class EnvironmentsMap {
-    public Map<String, Environments> environments;
-
-    enum Environments {
-      development
+    InputStream file;
+    if (Strings.isNullOrEmpty(configFile)) {
+      ClassLoader classLoader = SubzeroGui.class.getClassLoader();
+      URL resource = classLoader.getResource("sample.yaml");
```

Comment:
should we just make this mandatory to provide a --config?

Reply:
I like having the dev case just work.